templates/security/cve/_pagination.html

```diff
@@ -5,98 +5,114 @@
+          <a href="?{{ modify_query({'offset': (current_page - 5) * limit}) }}" class="p-pagination__link">{{ current_page - 4 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page > 3 and current_page == total_pages - 1 %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page - 4) * limit}) }}" class="p-pagination__link">{{ current_page - 3 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page > 3 and current_page == total_pages %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page - 4) * limit}) }}" class="p-pagination__link">{{ current_page - 3 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page > 2 %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page - 3) * limit}) }}" class="p-pagination__link">{{ current_page - 2 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page > 1 %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page - 2) * limit}) }}" class="p-pagination__link">{{ current_page - 1 }}</a>
+        </li>
+      {% endif %}
+  
+      {# current current_page #}
       <li class="p-pagination__item">
-        <a href="?{{ modify_query({'offset': (current_page + 3) * limit}) }}" class="p-pagination__link">{{ current_page + 4 }}</a>
+        <a href="?{{ modify_query({'offset': offset }) }}" class="p-pagination__link is-active">{{ current_page }}</a>
       </li>
-    {% endif %}
-
-    {% if current_page <= total_pages - 3 and current_page == 2 %}
+  
+      {% if current_page < total_pages %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': current_page * limit}) }}" class="p-pagination__link">{{ current_page + 1 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page < total_pages - 1 %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page + 1) * limit}) }}" class="p-pagination__link">{{ current_page + 2 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page < total_pages - 2 and current_page == 1 %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page + 2) * limit}) }}" class="p-pagination__link">{{ current_page + 3 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page < total_pages - 3 and current_page == 1 %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page + 3) * limit}) }}" class="p-pagination__link">{{ current_page + 4 }}</a>
+        </li>
+      {% endif %}
+  
+      {% if current_page <= total_pages - 3 and current_page == 2 %}
+        <li class="p-pagination__item">
+          <a href="?{{ modify_query({'offset': (current_page + 2) * limit}) }}" class="p-pagination__link">{{ current_page + 3 }}</a>
```

Comment:
this one looks exactly like the one 2 levels up. Are you sure the numbers are correct here?